### PR TITLE
kie-issues#167: Support List Data Types on inputs/outputs tables of DMN Runner

### LINKED ISSUE #167: Support `List` Data Types on inputs/outputs tables of DMN Runner
Based on what already works on the Form mode of DMN Runner, we need to support Lists on the Table mode as well.

The UX for supporting List types on the DMN Runner table is not clear yet, so part of this task is coming up with a proposal on how exactly we're going to do it.

__Idea 1__: We can use the same component we use for Boxed Expressions, with the possibility of collapsing/expanding it on demand, so that cells do not grow too much vertically.

```[tasklist]
### Acceptance criteria:
- [ ] Lists of any type should be supported on DMN Runner
- [ ] It should be possible to add/remove items of cells containing List inputs
- [ ] Lists should be rendered on the outputs table
```

### PR DESCRIPTION
**Pull request by ljmotta**
## issue
Closes https://github.com/kiegroup/kie-issues/issues/167

## On this PR
Added fully support to the List Data Types with keyboard shortcuts. Some new components were add to adapt the List Data Type in the table view:
 - Add UnitablesAutoFieldValue with:
   - Add UnitablesListField;
     - Add UnitablesListItemField;
   - Add UnitablesNestField;

## Video
https://github.com/kiegroup/kie-tools/assets/24302289/a5b1b1d1-0294-4b8a-9a4a-5c6e8ebe0287

### CODE REVIEW

**Comment by yesamer** [comment]
@ljmotta Please consider this scenario:
- Input Node: Type List of string 
- If I try to add an element, an error is thrown

https://github.com/kiegroup/kie-tools/assets/16005046/1965e8c0-19ff-4328-9b29-23ccf22a5209

**Comment by yesamer** [comment]
@ljmotta Another scenario: 
- Input Node: Type List of Structure (name, age)
- It's not possible to type any input with the keyboard


https://github.com/kiegroup/kie-tools/assets/16005046/a513dffc-9131-4735-bd02-3da5c17f26eb


https://github.com/kiegroup/kie-tools/assets/16005046/8e310e1a-73cb-412f-8556-39b9b667af1e

**Comment by yesamer** [comment]
It seems Lists are not supported in the case of Decisions (output columns). Is this expected?

https://github.com/kiegroup/kie-tools/assets/16005046/a7120e6e-61e4-402a-b6fc-6089ffb5c215

**Comment by yesamer** [comment]
With dozens of items, table css is misaligned 

Screenshot 2023-05-30 at 16 55 10

With an additional input column, is not possible to remove the last item

Screenshot 2023-05-30 at 17 00 30

**Comment by yesamer** [comment]
A List of Objects with a nested List field leads to an exception. 

https://github.com/kiegroup/kie-tools/assets/16005046/0472ff60-1ea7-4506-b7f5-2daac4bc2734

File: packages/online-editor/src/dmnRunner/DmnRunnerErrorBoundary.tsx [hunk 1]
```diff
@@ -50,9 +50,9 @@
             <TextContent>
               <I18nWrapped
                 components={{
-                  issues: (
-                    <a href={KIE_ISSUES_LINK} target={"_blank"}>
-                      {KIE_ISSUES_LINK}
+                  jira: (
+                    <a href={KOGITO_JIRA_LINK} target={"_blank"}>
```
[the remark this answers is not in the document]
Comment (yesamer) [reply packages/online-editor/src/dmnRunner/DmnRunnerErrorBoundary.tsx:54]: @ljmotta Most likely we can ignore this, but does make sense to add `noopener/noreferrer` as proposed?

**Comment by ljmotta** [comment]
@yesamer Thanks a lot for your review. I'll sumarize it here:

1. Simple lists are not working:
  Indeed, It's not working, I'll add a case for it.

2. It's not possible to type edit the cells:
  I've added a preventDefault for all keydown events 😢. I'll fix it

3. The Output column is not supported:
  It's the expected behavior. The outputs will change depedending of the inputs.

4. CSS missalignment with multiple items:
  Maybe it's a 1 to 2px miss aligment for each input in a object. Thanks for catching it!
  
5. List of Objects in a nested ListField throws a exception:
  I'm going to investigate it.

**Comment by yesamer** [comment]
@ljmotta You did an excellent job, the feature is quite complex, so not a surprise if some corner cases can fail. I'll continue to test it more deeply tomorrow.

**Comment by ljmotta** [comment]
@yeser I've addressed all reported issues:

1, 2 and 4:
https://github.com/kiegroup/kie-tools/assets/24302289/49cb196b-9c6a-4fb0-984e-88590e00f499


2,4 and 5:
https://github.com/kiegroup/kie-tools/assets/24302289/3a5b663d-5b4e-4b57-9b02-2243e66c1b17

**Comment by ljmotta** [comment]
@yesamer During some more tests, the `Load inputs` feature is not working correctly. I'll work on a fix for it.

**Review by tiagobento — APPROVED** [review]
Nice! Thanks @ljmotta!

**Comment by ljmotta** [comment]
This PR also closes [kie-issues#230](https://github.com/kiegroup/kie-issues/issues/230)

https://github.com/kiegroup/kie-tools/assets/24302289/ef9d36ad-2383-46ab-a2e7-bc7bd932a0c5
